python/feathub/feature_views/tests/test_feature_view.py

```diff
@@ -191,3 +192,47 @@ def test_duplicate_feature_names_throw_exception(self):
                 keep_source_fields=True,
             )
         self.assertIn("contains duplicated feature name", ctx.exception.args[0])
+
+    def test_get_output_fields(self):
+        field_names = ["id", "val1", "val2", "val3"]
+        source = DataGenSource(
+            name="source_1",
+            schema=Schema(field_names, [Int64, Int64, Int64, Int64]),
+            timestamp_field="lpep_dropoff_datetime",
+            timestamp_format="%Y-%m-%d %H:%M:%S",
+            keys=["id"],
+        )
+
+        feature_view_1 = DerivedFeatureView(
+            name="feature_view_1",
+            source=source,
+            features=["val2", "val3", "val1"],
+            keep_source_fields=True,
+        )
+
+        built_feature_view_1 = feature_view_1.build(self.registry)
+
+        if isinstance(built_feature_view_1, FeatureView):
+            self.assertEqual(
+                ["id", "val2", "val3", "val1"],
+                built_feature_view_1.get_output_fields(field_names),
+            )
+        else:
+            raise Exception("FeatureView is expected.")
```

Comment:
The Exception is to ensure that the test would fail if `built_feature_view_1` is not as expected. Adding the `if instanceOf...else...` is to avoid mypy check.